src/model/services/CompanyInfoService.php

```diff
@@ -0,0 +1,50 @@
+<?php 
+require_once 'src/model/repositories/CompanyInfoRepository.php';
+require_once 'src/model/services/AddressService.php';
+require_once 'src/model/entity/CompanyInfo.php';
+class CompanyInfoService {
+    private CompanyInfoRepository $companyInfoRepository;
+    private AddressService $addressService;
+    private PDO $db;
+
+    public function __construct() {
+        $this->db = $this->getdb();
+        $this->companyInfoRepository = new CompanyInfoRepository($this->db);
+        $this->addressService = new AddressService();
+    }
+
+    private function getdb(): PDO {
+        require_once 'src/model/database/dbcon/DatabaseConnection.php';
+        return DatabaseConnection::getInstance(); // singleton
+    }
+
+    /* Get company info */
+    public function getCompanyInfo(): array|CompanyInfo {
+        try {
+            $result = $this->companyInfoRepository->getCompanyInfo();
+            try {
+                $address = $this->addressService->getAddressById($result['addressId']);
+                return new CompanyInfo($result['companyInfoId'], $result['companyName'], $result['companyDescription'], $result['logoUrl'], $address);
+            } catch (Exception $e) {
+                return ["error"=> true, "message"=> $e->getMessage()];
+            }
+        } catch (PDOException $e) {
+            return ["error"=> true, "message"=> $e->getMessage()];
+        }
+    }
+
+    public function editCompanyInfo(int $companyInfoId, array $newCompanyInfoData): array|CompanyInfo {
+        $this->db->beginTransaction();
+        try {
+            $newAddress = new Address($newCompanyInfoData['addressId'], $newCompanyInfoData['street'], $newCompanyInfoData['streetNr'], 
+            new PostalCode($newCompanyInfoData['postalCodeId'], $newCompanyInfoData['postalCode'], $newCompanyInfoData['city']));
+            $this->addressService->editAddress($newCompanyInfoData['addressId'], $newAddress);
+            $this->companyInfoRepository->editCompanyInfo($companyInfoId, $newCompanyInfoData['companyName'], $newCompanyInfoData['companyDescription']);
+          $this->db->commit();
```

Comment:
So it gets commited only when the address and company info have been edited succefully without returning an error in the catch block?

Reply:
Exactly! And if any of the operations fail and we catch an error, we roll back everything.